utils/installer/install.sh

```diff
@@ -156,14 +206,22 @@ function check_system_deps() {
 }
 
 function install_nodejs_deps() {
-  check_dep "npm"
-  echo "Installing node modules with npm.."
-  for dep in "${__npm_deps[@]}"; do
-    if ! npm ls -g "$dep" &>/dev/null; then
+  check_dep "yarn" "npm"
+  if [ "$check_dep_return" == "npm" ]; then
```

Comment:
Using globals for means of communication is bad practice

Reply:
Thats the only thing you can do in the shell. No return values on the functions :(

Reply:
To 'return' values, you have to echo "value" and capture it back in the caller.

Anyway, since I think check_dep should be merged inside this function, it won't be an issue  anymore.
I have a very similar function on one of my projects if you want to draw inspiration:
https://github.com/Tastyep/Pi-OpenCast/blob/master/setup.sh#L30-L49

@kylo252 Are you ok with this ?

Reply:
I had an entire https://github.com/LunarVim/LunarVim/pull/1565#discussion_r711380560 about it where I concluded that we should get rid of it.

> Overall I actually think it's better to retain the original function since this is not gonna save us anything. It will be more clear if you just check each of them individually.

Reply:
I read your comment and agree with it.
In my comment I said we could go even further and I showed how I would do it.